src/cyclonedds/cyclonedds/pub.py

```diff
@@ -29,20 +31,36 @@
 class Publisher(Entity):
     def __init__(
             self,
-            domain_participant: 'cyclonedds.domain.DomainParticipant',
-            qos: Optional['cyclonedds.core.Qos'] = None,
-            listener: Optional['cyclonedds.core.Listener'] = None):
+            domain_participant: DomainParticipant,
+            qos: Optional[Qos] = None,
+            listener: Optional[Listener] = None):
+        if not isinstance(domain_participant, DomainParticipant):
+            raise TypeError(f"{domain_participant} is not a cyclonedds.domain.DomainParticipant or cyclonedds.sub.Subscriber.")
```

Comment:
Telling the user that he can also pass in a `cyclonedds.sub.Subscriber` seems weird?

Reply:
Copy-paste failure, will fix.